src/main/java/io/vertx/core/http/impl/HttpClientRequestImpl.java

```diff
@@ -57,12 +60,13 @@
 
   private static final Logger log = LoggerFactory.getLogger(HttpClientRequestImpl.class);
 
-  private final String host;
-  private final int port;
+  private String host;
+  private int port;
+  private int redirectCount;
   private final HttpClientImpl client;
-  private final HttpRequest request;
+  private HttpRequest request;
   private final VertxInternal vertx;
-  private final io.vertx.core.http.HttpMethod method;
+  private io.vertx.core.http.HttpMethod method;
```

Comment:
Need to be able to change that for HTTP 303 redirect where the method can be changed from POST to GET
https://en.wikipedia.org/wiki/HTTP_303